### DISCUSSION TITLE
How to use API to import OPML?

### DISCUSSION THREAD

**Discussion by Jorenar** [Athou/commafeed#1031] [Q&A] [answered]
I'm creating an UserScript which will sort the feeds alphabetically for me.
I'm trying to upload sorted OPML file with the following code:
```javascript
fetch(window.location.origin + '/rest/feed/import', {
  method : "POST",
  headers: { 'Content-Type': 'multipart/form-data' },
  body: new XMLSerializer().serializeToString(opml),
}).then(res => res.text()).then(console.log);
```
but I'm getting `{"code":400,"message":"HTTP 400 Bad Request"}`.

Adding header `Content-Disposition: form-data; name="file"; filename="rss.opml"` doesn't change anything;
using `Content-Type: text/xml` results in `{"code":415,"message":"HTTP 415 Unsupported Media Type"}`.

**Comment by Athou** [comment, marked as answer]
You need to post a `FormData` object and set the `file` param to the opml content. `file` is the name expected by CommaFeed for the opml in the POST body.

    const formData = new FormData();
    formData.append("file", opmlFile);

opmlFile is usually the value from the file picker but it seems you have the opml as xml or string. I think you'd need to construct a File yourself then: https://stackoverflow.com/a/26181292

After that you can set `formData` as the value for `body` in your `fetch` : `body: formData`.

See https://github.com/Athou/commafeed/blob/3271d69fcb5a829904f3d8db7ff939e730dfe2b9/commafeed-client/src/app/client.ts#L63-L72

**Reply by Jorenar** [reply]
Great! That works, thanks!
